TM1py/Services/ElementService.py

```diff
@@ -275,3 +276,21 @@ def get_members(element):
 
         get_members(consolidation_tree)
         return members
+
+    def execute_set_mdx(self, mdx, top=10000):
+        '''
+        :param mdx: valid dimension mdx statement
+        :param top: number of records to return
+        :return: dictionary of members, unique names, weights, types, ordinals, and parents
+        '''
+
+        request = f'/api/v1/ExecuteMDXSetExpression?$expand=Tuples($top={top};' \
+                  '$expand=Members($select=Name,Ordinal,Weight;' \
+                  '$expand=Parent($select=Name,UniqueName),' \
+                  'Element($select=Type,Level)))'
```

Comment:
python `f strings` are nice. We should be using them way more often in TM1py I think. 👍

Why ask for the Ordinal of the Member ? I think the ordinal is given implicitly since the Collections in the REST response are ordered.

In a future version of this we could perhaps build the url dynamically based on function arguments, so that the user could specify `member_properties`, `parent_properties` and `element_properties`. If no `parent_properties` are passed we wouldn't need to expand the parent then.

Reply:
Made the properties dynamic.